tests/gtest/avifdecodetest.cc

```diff
@@ -30,6 +33,14 @@ TEST(AvifDecodeTest, ColorGridAlphaNoGrid) {
   EXPECT_GT(decoder->image->alphaRowBytes, 0u);
 }
 
+TEST(AvifDecodeTest, ParseEmptyData) {
+  DecoderPtr decoder(avifDecoderCreate());
+  ASSERT_NE(decoder, nullptr);
+  ASSERT_EQ(avifDecoderSetIOMemory(decoder.get(), nullptr, 0), AVIF_RESULT_OK);
+  // No ftyp box was seen.
+  ASSERT_EQ(avifDecoderParse(decoder.get()), AVIF_RESULT_INVALID_FTYP);
```

Comment:
The test only allows the error code returned by the current implementation here, but we may want to give the implementation some flexibility to return any other reasonable error code in the future.

I guess what's important is that `avifDecoderParse()` should fail (the error code should not be `AVIF_RESULT_OK`). In addition, it would be good to check that the `avifIO::read()` call succeeds (`avifIO::read()` should not return `AVIF_RESULT_IO_ERROR`), but I don't know if we can reliably determine that from the return value of `avifDecoderParse()`.